onellm/files.py

```
@@ -49,24 +55,136 @@ def upload(
+                kwargs["filename"] = filename
+            
+        elif hasattr(file, "read"):
+            # File-like object - try to validate without reading entire file
+            
+            # Get filename from file.name or kwargs - required if allowed_extensions is set
+            filename = getattr(file, "name", None) or kwargs.get("filename")
+            if filename is None:
+                if allowed_extensions:
+                    raise InvalidRequestError(
+                        "filename parameter is required when uploading file-like object without .name attribute with allowed_extensions set"
+                    )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)